Add hands-on tutorial for go-dqlite (#11)
canonical/dqlite-docs: tutorials/setup-your-development-environment.md

> # Setup your development environment
> 
> The following section walks you through the necessary steps to setup your environment.

Reviewer: ```suggestion
# Set up your development environment

The following section walks you through the necessary steps to set up your environment.
```

It is recommended to spin up a clean Ubuntu VM for these steps as it is always good to perform experiments in an isolated development environment.

> The steps outlined [here](https://juju.is/docs/sdk/set-up-your-development-environment#heading--set-up-an-ubuntu-vm-with-multipass) explain how to do this.

Reviewer: ```suggestion
Follow the steps in [Set up an Ubuntu VM with Multipass](https://juju.is/docs/sdk/set-up-your-development-environment#heading--set-up-an-ubuntu-vm-with-multipass) to do this.
```
"Here" is a very bad link text, since users have no idea where they'll go, and Google doesn't like it either.

Author: Cool - thanks for the explanation. Especially, that Google doesn't like it was new to me.

## Create a project directory

Create a new project directory with
```bash
mkdir ~/go-dqlite-demo
cd ~/go-dqlite-demo
```

> This will create a new folder called `go-dqlite-demo` in your home directory and opens it. All project files will be put in this directory.

Reviewer: ```suggestion
This will create a new folder called `go-dqlite-demo` in your home directory and enters it. All project files will be put in this directory.
```

## Install the Dqlite tooling

> On a debian-based system you can get the latest development release from [Dqlite's development PPA](https://launchpad.net/~dqlite/+archive/ubuntu/dev). First add the PPA repository:

Reviewer: ```suggestion
On a Debian-based system, you can get the latest development release from [Dqlite's development PPA](https://launchpad.net/~dqlite/+archive/ubuntu/dev). First add the PPA repository:
```

```bash
sudo add-apt-repo ppa:dqlite/dev
sudo apt update
```

> then install the library with 

Reviewer: ```suggestion
Then install the library:
```

```bash
sudo apt install libdqlite-dev
```

[note type="important" status="Info"]

> If you are not using a debian-based system you’ll need to build libdqlite from source. See [this guide](https://github.com/canonical/dqlite#build) on instructions on how to do this.

Reviewer: ```suggestion
If you are not using a Debian-based system, you must build `libdqlite` from source. See [Build](https://github.com/canonical/dqlite#build) for instructions on how to do this.
```

[/note]

# Install Go 

> Download and install the latest version of Go following the steps outlined here: https://go.dev/doc/install.

Reviewer: ```suggestion
Download and install the latest version of Go following the steps outlined in the [Go documentation](https://go.dev/doc/install).
```

You can verify that everything is installed correctly by typing the following command in your terminal:

```bash
> go version

> go version go1.21.1 linux/amd64

Reviewer: We should give some information about what we expect to see here. "not found" would probably be bad. ;)
Maybe say that if the version is >something, it's good?

Author: There are no real version constrains. If the user follows the linked tutorial, the version should be pretty recent. 

I added one sentence that the user should check if the version output matches the one that was installed during the installation step.

```

In your project folder, create a new mod file with:

```bash
go mod init go-dqlite-demo
```

Install the application bindings for `go-dqilite/app` with:
```bash
go get github.com/canonical/go-dqlite/app
```

[note type="important" status="Info"]

> Read more about Go mod files here: https://go.dev/ref/mod

Reviewer: ```suggestion
Read more about Go mod files in the [Go documentation](https://go.dev/ref/mod).
```

[/note]

# Install a C compiler

> A C compiler is required for the package to build. Install `gcc` with

Reviewer: ```suggestion
A C compiler is required for the package to build. Install `gcc` with:
```

```bash
sudo apt install gcc
```

> You should now be all set to start your first Go application. Continue with [Create an example application](tbd)

Reviewer: ```suggestion
You should now be all set to start your first Go application. Continue with [Create an example application](tbd).
```
